index.js

```diff
@@ -249,3 +238,39 @@ module.exports = function prettyFactory (options) {
     }
   }
 }
+
+function asMetaWrapper (dest) {
+  const parsed = Symbol('parsedChindings')
+
+  if (!dest && !dest.write) {
+    throw new Error('the destination must be writable')
+  }
+
+  const pretty = this
+
+  return {
+    [Symbol.for('needsMetadata')]: true,
+    lastLevel: 0,
+    lastMsg: null,
+    lastObj: null,
+    lastLogger: null,
+    write (chunk) {
+      var chindings = this.lastLogger[parsed]
+
+      if (!chindings) {
+        chindings = JSON.parse('{"v":1' + this.lastLogger.chindings + '}')
+        this.lastLogger[parsed] = chindings
+      }
+
+      // TODO remove Object.assign as its slow
+      const obj = Object.assign({
```

Comment:
I tend to use https://www.npmjs.com/package/conflate when I'm looking for something other than `Object.assign`. See https://github.com/jsumners/merge-benchmarks

Reply:
probably we should just leave `Object.assign`, the performance is not that horrible compared to others.

Reply:
I am 👌 with `Object.assign`.